### PR TITLE
Tests: clean up test skip/requires checks (Trac 53009)

### PR DESCRIPTION
**Pull request by jrfnl**
Trac ticket: https://core.trac.wordpress.org/ticket/53009

See the individual commits for full details.

This is the result of a full review of the PHPUnit test suite for test requirements.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/image/functions.php [hunk 1]
```diff
@@ -339,6 +307,24 @@ public function test_load_directory() {
 		}
 	}
 
+	/**
```
Comment (jrfnl) [inline tests/phpunit/tests/image/functions.php:310]: @hellofromtonya Just a quick question: the `markTestSkipped()` check and call _could_ also be moved to this helper function. Was there any particular reason not to do so ?

Comment (hellofromtonya) [reply tests/phpunit/tests/image/functions.php:298]: @jrfnl Whoopsie, thought I had refactored it. Thanks for flagging it. Now resolved in commit https://github.com/WordPress/wordpress-develop/pull/1218/commits/f97d2501f8a9e7b85d48a3b575e55073df5d8587

**Comment by jrfnl** [comment]
FYI: CS issue is unrelated to this PR, but caused by a CS issue introduced in commit 955df634d1b4899008bd5900d99c02f773d76e66

**Comment by jrfnl** [comment]
Closing as committed via https://core.trac.wordpress.org/changeset/51415

**Pull request closed without merging**